### PR TITLE
[NUI] Fix StyleCop warning CA2227

### PR DESCRIPTION
**Pull request by jaehyun0cho**
To fix CA2227(Collection properties should be read only), the setters of
the collection properties are removed.

### API Changes ###

 - ACR:

### CODE REVIEW

**Comment by jaehyun0cho** [comment]
The setter of the collection property URLS in AnimatedImageVisual.cs is not removed because URLS has been publicly opened since tizen API version 4.

File: src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs [hunk 1]
```diff
@@ -33,11 +33,18 @@ public class Selector<T>
     {
         private readonly bool cloneable = typeof(ICloneable).IsAssignableFrom(typeof(T));
 
+        private IList<StateValuePair<T>> stateValueList = new List<StateValuePair<T>>();
+
         /// <summary>
         /// The list for adding state-value pair.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public IList<StateValuePair<T>> StateValueList { get; set; } = new List<StateValuePair<T>>();
```
Comment (dasg34) [inline src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs:40]: Oh, `public` is temporarily added to fix DA build error. (https://github.com/Samsung/TizenFX/pull/2142)

You don't need to remove setter of `StateValueList`.

**Review by dasg34 — CHANGES_REQUESTED** [review]
Do not remove `IResourcesProvider.XamlResources` setter.

`XamlResources` is necessary for using `ResourceDictionary` in xaml.

For example,
```xaml
<View x:Class="NUIXamlTemplate1.XamlPage"
  xmlns="http://tizen.org/Tizen.NUI/2018/XAML"
  xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" >

    <View.XamlResources>
      <TextLabelStyle x:Name="TextLabelStyle1" x:Key="TextLabelStyleKey" TextColor="Yellow">
      </TextLabelStyle>
    </View.XamlResources>
    ...
    ...
    <View.Text Style="{x:StaticResource TextLabelStyle1}"/>
</View>
```

File: src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs (hunk 1 above)
Comment (jaehyun0cho) [reply src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs]: Thank you for your comment!
I have restored the code.

**Comment by jaehyun0cho** [comment]
@dasg34 
Thank you for your comments!
I have updated the patch not to remove setter of XamlResources.

**Review by AdunFang — APPROVED** [review]
It is OK.

**Review by dasg34 — APPROVED** [review]
Thank you for the patch~
Could you rebase this patch?
there is conflict with formatting patch(https://github.com/Samsung/TizenFX/commit/d9c0d98ce78a7a36509805aeb19ce0584b225215)

**Comment by dasg34** [comment]
I could resolve conflict on github web :)

**Pull request merged**